viewing_party/party.py

```diff
@@ -1,23 +1,166 @@
-# ------------- WAVE 1 --------------------
+from pprint import pprint
+#HELPER FUNCTION
+def _get_movies_watched_by_friends(user_data):
+    movies_watched_by_friends = []
+    
+    for friend in user_data["friends"]: #[{}]
+        for friend_movie in friend["watched"]: #{}
+            if friend_movie in movies_watched_by_friends:
+                continue
+            else:
+                movies_watched_by_friends.append(friend_movie)
+
+    return movies_watched_by_friends
+
+def _get_movies_not_watched_user(user_data):
+    movies_not_watched_user = []
+    movies_watched_by_friends = _get_movies_watched_by_friends(user_data) 
+
+    for movie in user_data["watched"]:
+        if movie not in movies_watched_by_friends:
+            movies_not_watched_user.append(movie)
+    return  movies_not_watched_user
+
 
+# ------------- WAVE 1 --------------------
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None 
+
+    return {"title":title, "genre":genre, "rating":rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        watchlist_title = movie["title"]
+        if title == watchlist_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+
+    return user_data
+# Note: For Waves 2, 3, 4, and 5, your implementation of each of the functions should not modify user_data.
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = 0
+    num_of_movies = len(user_data["watched"])
+    
+    for movie in user_data["watched"]:
+        rating += movie["rating"]
+
+    if rating == 0:

```

Comment:
Great job with this method! Couple of small callouts here.

You could convey intent a little more explicitly by pulling the logic around `return 0` into it's own guard clause with (on line 57 so it's the first line of code that executes in this method):
```python
if not user_data["watched"]:
      return 0
```
then proceed to build up your rating with the for-loop.

This way, when another developer comes along, I can immediately recognize "Oh if this value doesn't exist, then we'll return 0. The way your code is currently structured, I can tell at some point this method will return 0, but it's not as immediately clear.

Also a note about guard clauses - The Guard Clause pattern helps in these cases by “inverting” that way of thinking and rapidly returning false conditions and keeping the important code blocks (usually the longest) in the lowest indentation level possible